nixos/modules/config/users-groups.nix

```diff
@@ -663,6 +677,20 @@ in {
       '';
     };
 
+    system.activationScripts.update-lingering = let
+      lingerDir = "/var/lib/systemd/linger";
+      lingeringUsers = map (u: u.name) (attrValues (flip filterAttrs cfg.users (n: u: u.linger)));
+      lingeringUsersFile = builtins.toFile "lingering-users"
+        (concatStrings (map (s: "${s}\n")
+          (sort (a: b: a < b) lingeringUsers)));  # this sorting is important for `comm` to work correctly
+    in stringAfter [ "users" ] ''
+      if [ -e ${lingerDir} ] ; then
+        cd ${lingerDir}
+        ls ${lingerDir} | sort | comm -3 -1 ${lingeringUsersFile} - | xargs -r ${pkgs.systemd}/bin/loginctl disable-linger
```

Comment:
This wouldn't deactivate lingering for a user after it is was enabled and disabled again, right? It would just flip lingering for all users that have it enabled.

Because when I updated my system today I noticed that I have a new empty lingering-user file

Reply:
Fixing the empty file behavior in #261110

Reply:
Sorry, I'm just a random person that followed the issue where this commit was first introduced, and I may be missing a bunch of things here, but wouldn't relying on mutableUsers in turn break the ability of setting this through configuration.nix for users that aren't mutable?

Reply:
does this "break" (i'm using this term very lightly) my system if i remove a user? i'm thinking yes because the `update-lingering` activation script keeps failing on me - consider
```
$ sudo loginctl disable-linger someone-who-no-longer-exists                         nix-shell-env
Failed to look up user someone-who-no-longer-exists: No such process
```

my system works fine if i simply `rm` the offending file out of `/var/lib/systemd/linger`

cc @ToxicFrog @SuperSandro2000 @ambroisie

Reply:
as mentioned in https://github.com/NixOS/nixpkgs/issues/283769#issuecomment-1964266115 i'm considering reverting this PR - would anyone care to comment before i do that?